uint8 control_switch
uint8 power_switch